pkg/ring/lifecycler_delegates.go

```diff
@@ -0,0 +1,28 @@
+package ring
+
+import (
+	"time"
+
+	"github.com/go-kit/log"
+)
+
+// AutoForgetDelegate automatically remove an instance from the ring if the last
+// heartbeat is older than a configured period.
+type LifecyclerAutoForgetDelegate struct {
+	next         LifecyclerDelegate
+	logger       log.Logger
+	forgetPeriod time.Duration
+}
+
+func NewLifecyclerAutoForgetDelegate(forgetPeriod time.Duration, next LifecyclerDelegate, logger log.Logger) *LifecyclerAutoForgetDelegate {
+	return &LifecyclerAutoForgetDelegate{
+		next:         next,
+		logger:       logger,
+		forgetPeriod: forgetPeriod,
+	}
+}
+
+func (d *LifecyclerAutoForgetDelegate) OnRingInstanceHeartbeat(lifecycler *Lifecycler, ringDesc *Desc) {
+	AutoForgetFromRing(ringDesc, d.forgetPeriod, d.logger)
+	d.next.OnRingInstanceHeartbeat(lifecycler, ringDesc)
```

Comment:
As we know there is not really a `next` delegate doing anything, I feel we can safely remove `next` here. I don't see a usecase to really chain this here for now.

Reply:
I would leave this next here @yeya24 
It will be painful if someone add a new delegate and try to debug why it is not working. Easier just follow the pattern. We can refactor all these delegates when we are joining both lifecycler into to one